x-pack/legacy/plugins/siem/public/components/inspect/index.tsx

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiButtonEmpty, EuiButtonIcon } from '@elastic/eui';
+import { getOr } from 'lodash/fp';
+import React from 'react';
+import { connect } from 'react-redux';
+import { ActionCreator } from 'typescript-fsa';
+import styled from 'styled-components';
+
+import { pure } from 'recompose';
+import { inputsModel, inputsSelectors, State } from '../../store';
+import { InputsModelId } from '../../store/inputs/constants';
+import { inputsActions } from '../../store/inputs';
+
+import { ModalInspectQuery } from './modal';
+import * as i18n from './translations';
+
+const InspectContainer = styled.div<{ showInspect: boolean }>`
+  .euiButtonIcon {
+    ${props => (props.showInspect ? 'opacity: 1;' : 'opacity: 0')}
+    transition: opacity ${props => getOr(250, 'theme.eui.euiAnimSpeedNormal', props)} ease;
+  }
+`;
+
+interface OwnProps {
+  queryId: string;
+  inputId?: InputsModelId;
+  inspectIndex?: number;
+  isDisabled?: boolean;
+  onCloseInspect?: () => void;
+  show: boolean;
+  title: string | React.ReactElement | React.ReactNode;
+}
+
+interface InspectButtonReducer {
+  id: string;
+  isInspected: boolean;
+  loading: boolean;
+  inspect: inputsModel.InspectQuery | null;
+  selectedInspectIndex: number;
+}
+
+interface InspectButtonDispatch {
+  setIsInspected: ActionCreator<{
+    id: string;
+    inputId: InputsModelId;
+    isInspected: boolean;
+    selectedInspectIndex: number;
+  }>;
+}
+
+type InspectButtonProps = OwnProps & InspectButtonReducer & InspectButtonDispatch;
+
+const InspectButtonComponent = pure<InspectButtonProps>(
```

Comment:
I would use `React.memo` and not `pure` anymore for all new code and when we touch or refactor code I would recommend replacing `pure` with `React.memo`

Reply:
`React.memo` won't work with `redux connect`, so I had to go old school.

Reply:
Also, I would like to upgrade our react-redux so we can use their new hooks so we can use all the latest without a problem.

Reply:
I see, thanks for the update on that. That's good to know.